src/stream.ts

```diff
@@ -179,7 +205,25 @@ export class WebRTCStream implements Stream {
    * Call when a remote error occurs, should close the stream for reading and writing
    */
   reset(): void {
-    this.close();
+    this._sendFlag(pb.Message_Flag.RESET);
     this.stat = defaultStat(this.stat.direction);
+    this.writeClosed = true;
+    this.closeWritePromise.resolve();
+    if (this.readClosed && this.writeClosed) {
+      this.close();
+    }
+  }
+
+  private _sendFlag(flag: pb.Message_Flag): void {
+    if (this.writeClosed) {
```

Comment:
we should still be able to send a flag if write is closed. This is useful in case of stream resets.